src/wallet/KintoWallet.sol

```diff
@@ -135,17 +135,26 @@ contract KintoWallet is Initializable, BaseAccount, TokenCallbackHandler, IKinto
      * @param newPolicy new policy
      */
     function setSignerPolicy(uint8 newPolicy) public override onlySelf {
+        // reverting to SingleSigner is not allowed for security reasons
+        if (newPolicy == signerPolicy || newPolicy == SINGLE_SIGNER) {
```

Comment:
should we move this check to _setSignerPolicy that way is not repeated between the two functions?

Reply:
No, I strongly oppose that, because all checks should be conducted at the beginning of external functions. The problem with performing checks within internal functions that are reused is that these functions may eventually be relocated, and the conditions may no longer be verified. In fact, I advocate for moving all checks to the beginning of external functions. In a way, each external function should offer its own independent security guarantees. For example, setSignerPolicy and resetSigners operate under slightly different assumptions.